application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -75,6 +75,22 @@
   #set ($keyboardShortcutsEnabled = false)
   #set ($discard = $xwiki.ssx.use('Diagram.DiagramSheet'))
   #set ($discard = $xwiki.jsx.use('Diagram.DiagramEditSheet'))
+  ## Issue #219: Cannot modify diagram title in diagram editor
+  &lt;div class="row xform"&gt;
+    &lt;div class="col-xs-12 col-md-12"&gt;
+      &lt;dl&gt;
+        &lt;dt&gt;
+          &lt;label for="xwikiDocTitleInput"&gt;
+            $services.localization.render('core.editors.content.titleField.label')
+          &lt;/label&gt;
+        &lt;/dt&gt;
+        &lt;dd&gt;
+          &lt;input id="xwikiDocTitleInput" type="text" name="title" value="$escapetool.xml($doc.title)"&gt;
```

Comment:
I know it doesn't make much sense to translate structured pages, such as diagram pages, but it is possible. Sure, all translations will share the same diagram object but they can have different titles, so it's best to edit the title corresponding to the current translation:

```suggestion
          &lt;input id="xwikiDocTitleInput" type="text" name="title" value="$escapetool.xml($tdoc.title)"&gt;
```

``tdoc`` stands for "translated" document.